activity_calendar/api.py

```diff
@@ -69,3 +73,53 @@ def fullcalendar_feed(request):
             activity_moment_jsons.append(json_instance)
 
     return JsonResponse({'activities': activity_moment_jsons})
+
+
+@require_safe
+def upcoming_core_feed(request):
+    """
+        A collection of the first occurrence of all core activities. E.g. the first
+        upcoming boardgame-related activity, the first upcoming swordfighting training, etc.
+        Skips cancelled and removed activities.
+        Used on kotkt.nl
+    """
+    grouping_identifiers = request.GET.get('groups', "").split(',')
+
+    now = timezone.now()
+
+    # Add placeholders to ensure the order of the groupings does not change
+    earliest_moments = {
+        x:None for x in grouping_identifiers
+    }
+
+    # Iterate over all published activities that are part of the core groupings in the GET request
+    for activity in Activity.objects.select_related('core_grouping').filter(published_date__lte=now, core_grouping__identifier__in=grouping_identifiers):
+        # Fetch the next activitymoment (if any) for this activity that is not cancelled
+        activity_moment = activity.get_next_activitymoment(dtstart=now, exclude_cancelled=True)
```

Comment:
To futureproof this a bit more (just in case the default of get_next changes). Also, it looks nicer as you already differ from exclude_cancelled default.
```suggestion
        activity_moment = activity.get_next_activitymoment(dtstart=now, exclude_cancelled=True, exclude_removed=True)
```